File: Medium/Python/435. Non-overlapping Intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        intervals.sort(key = lambda x : x[1])
        count = 0
        prev = 0
        
        for i in range(1, len(intervals)):
            if intervals[i][0] < intervals[prev][1]:
                count += 1
            else:
                prev = i
        
        return count

File: Medium/Java/435. Non-overlapping Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[1] - b[1]);
        int count = 0;
        int prev = 0;

        for(int i = 1; i < intervals.length; i++){
            if(intervals[i][0] < intervals[prev][1])
                count++;
            else
                prev = i;
        }

        return count;
    }
}
